maps/src/main/java/com/gluonhq/impl/maps/TileImageView.java

```diff
@@ -68,12 +69,16 @@ public TileImageView(int zoom, long i, long j) {
             }).thenAccept(file -> {
                 logger.fine("Tile from downloaded file " + zoom + "/" + i + "/" + j);
                 downloading.setValue(false);
-                setImage(file);
+                Platform.runLater(() -> {
+                    setImage(file);
+                });
                 setProgress(1);
             });
         } else {
             logger.fine("Tile from file cache");
-            setImage(future.getNow(null));
+            Platform.runLater(() -> {
```

Comment:
I don't think this one es needed, only the block in `future.thenAccept(file -> {})`  runs outside the JavaFX Application thread.